tests/validations/test_rtree_valid_check.py

```diff
@@ -26,3 +26,35 @@ def test_with_gpkg_allcorrect():
     dataset = open_dataset("tests/data/test_allcorrect.gpkg")
     checks = list(rtree_valid_check_query(dataset))
     assert len(checks) == 0
+
+
+def test_with_gdal_error():
+    results = []
+
+    # Register GDAL error handler function
+    def gdal_error_handler(err_class, err_num, error):
+        results.append("GDAL_ERROR")
+
+    init_gdal(gdal_error_handler)
+
+    dataset = open_dataset("tests/data/test_gdal_error.gpkg")
+    validations = dataset.ExecuteSQL('select rtreecheck("rtree_table_geom");')
+    dataset.ReleaseResultSet(validations)
+    assert results[0] == "GDAL_ERROR"
+    assert len(results) == 1
+
+
+def test_without_gdal_error():
```

Comment:
Ik dacht aan nog een:

```
def test_without_gdal_error():
    results = []

    # Register GDAL error handler function
    def gdal_error_handler(err_class, err_num, error):
        results.append("GDAL_ERROR")

    init_gdal(gdal_error_handler)

    dataset = open_dataset("tests/data/test_gdal_error.gpkg")
    with silence_gdal():
        validations = dataset.ExecuteSQL('select rtreecheck("rtree_table_geom");')
    dataset.ReleaseResultSet(validations)
    validations2 = dataset.ExecuteSQL('select rtreecheck("rtree_table_geom");')
    dataset.ReleaseResultSet(validations2)
    assert len(results) == 1
    assert results[0] == "GDAL_ERROR"
```

Reply:
Puur omdat ik zelf nog ff wilde weten of:

- alles goed gaat qua functiescopes (vandaar die functies elders)
- een slicence goedgaat tussen andere errors. 

(Maar misschien is dit ook een beetje overkill)

Reply:
En misschien ga ik hier ook de mist in qua unittesten (wat is de unit ;) )